src/semgrep_agent/findings.py

```diff
@@ -163,6 +164,52 @@ def to_gitlab(self) -> Dict[str, Any]:
             ],
         }
 
+    def to_gitlab_secrets(self) -> Dict[str, Any]:
+
+        return {
+            # IDs need to be persistent across findings for the same issue
+            # This needs to account for the same rule matching in different
+            # files and the same file -- even on the same syntactic value
+            "id": str(self.syntactic_identifier_int()),
```

Comment:
You might want to use syntactic_identifier_str if it needs to be a string.

```suggestion
            "id": self.syntactic_identifier_str(),
```